primitives/src/locktime/relative.rs

```diff
@@ -193,29 +212,42 @@ impl LockTime {
     #[inline]
     pub const fn is_block_time(self) -> bool { !self.is_block_height() }
 
-    /// Returns true if this [`relative::LockTime`] is satisfied by either height or time.
-    ///
+    /// Returns true if this [`relative::LockTime`] is satisfied by the given chain state.
     /// # Examples
     ///
     /// ```rust
-    /// # use bitcoin_primitives::Sequence;
-    /// # use bitcoin_primitives::relative;
-    ///
-    /// # let required_height = 100;       // 100 blocks.
-    /// # let intervals = 70;     // Approx 10 hours.
-    /// # let current_height = || relative::Height::from(required_height + 10);
-    /// # let current_time = || relative::Time::from_512_second_intervals(intervals + 10);
-    /// # let lock = Sequence::from_height(required_height).to_relative_lock_time().expect("valid height");
-    ///
-    /// // Users that have chain data can get the current height and time to check against a lock.
-    /// assert!(lock.is_satisfied_by(current_height(), current_time()));
+    /// # use bitcoin_primitives::locktime::relative::Height;
+    /// # use bitcoin_primitives::relative::Time;
+    /// # use units::mtp_height::MtpAndHeight;
+    /// # use bitcoin_primitives::BlockHeight;
+    /// # use bitcoin_primitives::relative::LockTime;
+    /// # use bitcoin_primitives::BlockTime;
+    ///
+    /// fn generate_timestamps(start: u32, step: u16) -> [BlockTime; 11] {
+    ///     let mut timestamps = [BlockTime::from_u32(0); 11];
+    ///     for (i, ts) in timestamps.iter_mut().enumerate() {
+    ///         *ts = BlockTime::from_u32(start.saturating_sub((step * i as u16).into()));
+    ///     }
+    ///     timestamps
+    /// }
+    /// // time extracted from BlockHeader
+    /// let timestamps: [BlockTime; 11] = generate_timestamps(1_600_000_000, 200);
+    /// let utxo_timestamps: [BlockTime; 11] = generate_timestamps(1_599_000_000, 200);
+    ///
+    /// let current_height = BlockHeight::from_u32(100);
+    /// let utxo_height = BlockHeight::from_u32(80);
+    ///
+    /// let chain_tip = MtpAndHeight::new(current_height, timestamps);
+    /// let utxo_mined_at = MtpAndHeight::new(utxo_height, utxo_timestamps);
+    /// let locktime = LockTime::Time(Time::from_512_second_intervals(10));
+    ///
+    /// // Check if locktime is satisfied
+    /// assert!(locktime.is_satisfied_by(chain_tip, utxo_mined_at));
```

Comment:
This looks very similar/same as above. Perhaps we should have a common example somewhere and just have links to it.